### DISCUSSION TITLE
How To: Add a hash to our public/assets files names (ex: images)?

### DISCUSSION THREAD

**Discussion by djimnz** [nuxt/nuxt#22529] [Questions] [answered]
Hello everyone,

Working on a landing page. I have added a Nitro config for my images Cache-Control that doesn't always change.
But I would like to add a hash to the file names such as: image-[hash].png

This would help my users to catch them if the build change even tho the Cache-Control was setup for many days.

Do you have an idea of witch configuration I should use under my nuxt config?
I'm not sure if this has to be done under Vite, or Nitro, or... (quite confusing)

Thank you!

**Comment by logotip4ik** [comment, marked as answer]
Hello!

I managed to do so via `vite.build.rollupOptions.output.assetFileNames` ([Rollup docs](https://rollupjs.org/configuration-options/#output-assetfilenames)). For example if you want to include just hash in your output filenames you could do something like this:

```js
// nuxt.config.ts
export default defineNuxtConfig({
  vite: {
    build: {
      rollupOptions: {
        output: {
          assetFileNames: '[hash].[ext]',
        },
      },
    },
  },
})
```

For more granular control you can provide function for `assetsFileNames`

**Reply by djimnz** [reply]
Hi @logotip4ik !

Thank you for your answer, it did the work. After reading your comment, I checked where my image was located. And that image was in the public directory instead of assets. So it couldn't get handled by Vite.

After moving it to the correct place, I could use it with `<img src="~/assets/images/..." />`!
But it seems like this is not compatible with nuxt-image.

Problem solved, if not using nuxt-image :)
